### ISSUE TITLE
azure - function policy deploy should be idempotent

### ISSUE THREAD

**Issue by kapilt** [cloud-custodian/cloud-custodian#5427]
**Is your feature request related to a problem? Please describe.**

if i redeploy a policy with no changes, i expect a no-op update. not a full deployment, ie. this should take seconds to provision, not minutes. the function zip should be checked to see if its the same as what's been uploaded before and continue rather than initiating a deploy again.

**Describe the solution you'd like**

the function zip should be checked to see if its the same as what's been uploaded (checksum) before and continue rather than initiating a deploy again.

**Comment by kapilt** [comment]
the underlying ftp/website scm octopus underneath functions https://github.com/projectkudu/kudu/wiki/REST-API

it might be easiest to just upload a json file with the checksum directly into the vfs , since none of the kudu apis support any notion of content checksum 🤷‍♂   alternatively tag the function.

**Comment by kapilt** [comment]
its not clear which of those apis is available in kudulite for linux consumption plans.

**Comment by logachev** [comment]
Agreed.. I asked for the checksum feature, but there is not any eta.

In the meantime I'm considering uploading checksum to the blob storage ourselves..

**Comment by logachev** [comment]
re: tag.. Didn't think about it.. Probably it is better than the blob
